aidial_adapter_bedrock/llm/model/llama/v3.py

```diff
@@ -1,72 +1,40 @@
-"""
-Turning a chat into a prompt for the Llama3 model.
+import json
 
-See as a reference:
-https://github.com/meta-llama/llama3/blob/main/llama/tokenizer.py
-"""
-
-from typing import List, Literal, Optional, Tuple, assert_never
-
-from aidial_adapter_bedrock.llm.chat_emulator import ChatEmulator
-from aidial_adapter_bedrock.llm.message import (
-    AIRegularMessage,
-    BaseMessage,
-    HumanRegularMessage,
-    SystemMessage,
+from aidial_adapter_bedrock.dial_api.request import ModelParameters
+from aidial_adapter_bedrock.llm.converse.adapter import (
+    ConverseAdapter,
+    ConverseMessages,
 )
-from aidial_adapter_bedrock.llm.model.llama.conf import LlamaConf
-
-
-def get_role(message: BaseMessage) -> Literal["system", "user", "assistant"]:
-    match message:
-        case SystemMessage():
-            return "system"
-        case HumanRegularMessage():
-            return "user"
-        case AIRegularMessage():
-            return "assistant"
-        case _:
-            assert_never(message)
-
-
-def encode_header(message: BaseMessage) -> str:
-    ret = ""
-    ret += "<|start_header_id|>"
-    ret += get_role(message)
-    ret += "<|end_header_id|>"
-    ret += "\n\n"
-    return ret
-
-
-def encode_message(message: BaseMessage) -> str:
-    ret = encode_header(message)
-    ret += message.text_content.strip()
-    ret += "<|eot_id|>"
-    return ret
-
-
-def encode_dialog_prompt(messages: List[BaseMessage]) -> str:
-    ret = ""
-    ret += "<|begin_of_text|>"
-    for message in messages:
-        ret += encode_message(message)
-    ret += encode_header(AIRegularMessage(content=""))
-    return ret
+from aidial_adapter_bedrock.llm.converse.types import (
+    ConverseDeployment,
+    ConverseRequestWrapper,
+)
+from aidial_adapter_bedrock.llm.tokenize import default_tokenize_string
 
 
-class LlamaChatEmulator(ChatEmulator):
-    def display(self, messages: List[BaseMessage]) -> Tuple[str, List[str]]:
-        return encode_dialog_prompt(messages), []
+class ConverseStreamingEmulateAdapter(ConverseAdapter):
+    """
+    Llama 3.1 supports tools, but only in non-streaming mode.
```

Comment:
Not only `Llama 3.1`. 
Make the statement more generic: `Certain Llama 3 models support tools only in the non-streaming mode`.

Reply:
Done, but removed this prefix `certain` 
Currently *all* Llama models do not support streaming for tools
If I would leave this *certain*, it would be natural to go and search for some models, that do.